### ISSUE TITLE
[coor.data.NumPyFileReader] FileNotFoundError: .pyemma/trajlen_cache.dir

### ISSUE THREAD

**Issue by gph82** [markovmodel/PyEMMA#775]
Got this. (takes about 30 min to reproduce, it's what I am doing now) 

In the meantime, perhaps this is known to somebody. Note how in "dumb.py" some python3.4 dev has commented already "may raise KeyError" and indeed the error appears. Is the the error upstream? 

All files in "npy_files" exist and have the same number of "columns"

``` python
pyemma.__version__
'2.1-14-g54a1a42'

Obtaining file info:  20% ( 2956/14486) [####                    ] eta 1:21:33 |Traceback (most recent call last):
  File "/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/util/traj_info_cache.py", line 204, in __getitem__
    result = str(self._database[key])
  File "/home/mi/gph82/miniconda3/lib/python3.4/dbm/dumb.py", line 131, in __getitem__
    pos, siz = self._index[key]     # may raise KeyError
KeyError: b'-2106045956875000278'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "FAH_scripts/get_TICA_numpy_filereader.py", line 24, in <module>
    my_npy_reader = coor.data.NumPyFileReader(npy_files)
  File "/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/numpy_filereader.py", line 60, in __init__
    self.filenames = filenames
  File "/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/_base/datasource.py", line 115, in filenames
    info = TrajectoryInfoCache.instance()[filename, self]
  File "/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/util/traj_info_cache.py", line 215, in __getitem__
    result = self.__setitem__(filename, info)
  File "/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/util/traj_info_cache.py", line 263, in __setitem__
    self._database[str(traj_info.hash_value)] = dbval
  File "/home/mi/gph82/miniconda3/lib/python3.4/dbm/dumb.py", line 181, in __setitem__
    self._addkey(key, self._addval(val))
  File "/home/mi/gph82/miniconda3/lib/python3.4/dbm/dumb.py", line 167, in _addkey
    self._chmod(self._dirfile)
  File "/home/mi/gph82/miniconda3/lib/python3.4/dbm/dumb.py", line 260, in _chmod
    self._os.chmod(file, self._mode)
FileNotFoundError: [Errno 2] No such file or directory: '/home/mi/gph82/.pyemma/trajlen_cache.dir'
```

**Comment by franknoe** [comment]
I think we got to get rid of the default write behavior asap and release 
a patch. This is causing too many problems.

Am 18/04/16 um 11:24 schrieb Guillermo Pérez-Hernández:

> Got this. (takes about 30 min to reproduce, it's what I am doing now)
> 
> In the meantime, perhaps this is known to somebody. Note how in 
> "dumb.py" some python3.4 dev has commented already "may raise 
> KeyError" and indeed the error appears. Is the the error upstream?
> 
> All files in "npy_files" exist and have the same number of "columns"
> 
> pyemma.**version**
> '2.1-14-g54a1a42'
> 
> Obtainingfile  info:20%  (2956/14486) [#### ] eta 1:21:33 |Traceback (most recent call last):
>    File"/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/util/traj_info_cache.py", line204,in  **getitem**
>      result =str(self._database[key])
>    File"/home/mi/gph82/miniconda3/lib/python3.4/dbm/dumb.py", line131,in  __getitem__
>      pos, siz =self._index[key]# may raise KeyError
> KeyError:b'-2106045956875000278'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>    File"FAH_scripts/get_TICA_numpy_filereader.py", line24,in  <module>
>      my_npy_reader = coor.data.NumPyFileReader(npy_files)
>    File"/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/numpy_filereader.py", line60,in  **init**
>      self.filenames = filenames
>    File"/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/_base/datasource.py", line115,in  filenames
>      info = TrajectoryInfoCache.instance()[filename,self]
>    File"/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/util/traj_info_cache.py", line215,in  __getitem__
>      result =self.**setitem**(filename, info)
>    File"/home/mi/gph82/programs/PyEmma/pyemma/coordinates/data/util/traj_info_cache.py", line263,in  **setitem**
>      self._database[str(traj_info.hash_value)] = dbval
>    File"/home/mi/gph82/miniconda3/lib/python3.4/dbm/dumb.py", line181,in  __setitem__
>      self._addkey(key,self._addval(val))
>    File"/home/mi/gph82/miniconda3/lib/python3.4/dbm/dumb.py", line167,in  _addkey
>      self._chmod(self._dirfile)
>    File"/home/mi/gph82/miniconda3/lib/python3.4/dbm/dumb.py", line260,in  _chmod
>      self._os.chmod(file,self._mode)
> FileNotFoundError: [Errno2] No suchfile  or  directory:'/home/mi/gph82/.pyemma/trajlen_cache.dir'
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub 
> https://github.com/markovmodel/PyEMMA/issues/775

## 

---

Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de

## Mail:  Arnimallee 6, 14195 Berlin, Germany

**Comment by gph82** [comment]
could not reproduce in a second run...I am putting this on hold

**Comment by marscher** [comment]
Am 18.04.2016 um 11:26 schrieb Frank Noe:

> I think we got to get rid of the default write behavior asap and release
> a patch. This is causing too many problems.
> The PR for this is already open, but I need to add some final changes. 
> Expect a merge and release soon.

**Comment by franknoe** [comment]
thanks

Am 18/04/16 um 13:50 schrieb Martin K. Scherer:

> Am 18.04.2016 um 11:26 schrieb Frank Noe:
> 
> > I think we got to get rid of the default write behavior asap and release
> > a patch. This is causing too many problems.
> > The PR for this is already open, but I need to add some final changes.
> > Expect a merge and release soon.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> https://github.com/markovmodel/PyEMMA/issues/775#issuecomment-211344638

## 

---

Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de

## Mail:  Arnimallee 6, 14195 Berlin, Germany

**Comment by marscher** [comment]
will be fixed upon merge of #774

**Comment by gph82** [comment]
confirm that this issue is no more
